src/math/CTiglPointTranslator.cpp

```diff
@@ -161,6 +161,7 @@ TiglReturnCode CTiglPointTranslator::translate(const CTiglPoint& p, double* eta,
     }
     
     assert(initialized);
+    SegmentProjection projector(*this, a, b, c, d);
```

Comment:
Did you evaluate the performance penalty? I chose to initialize the projector once, the get the maximum performance here.

Reply:
I just checked in the CI. The runtime of the function goes up from 1.08 us to about 1.29 us which is a noticable performance regression.
Are the CTiglPointTranslator changes required, to make the caching work / to make the class thread safe? Would the be a workaround?

Reply:
I believe I had to change it to gain const correctness and to allow multiple threads to concurrently call CTiglPointTranslator::translate

Reply:
![image](https://user-images.githubusercontent.com/1224051/43766281-5bbdf4c2-9a32-11e8-96ea-bdcbb609c021.png)
i quickly did a profile, the creation of the SegmentProjection object should be negligible (~1.23% runtime).

Reply:
In this case, leave it as it is :)

Reply:
Btw... What tool do you use for profiling?

Reply:
Visual Studio 2017's built in profiler
Menu: Analyze -> Performance Profiler ...